src/ai/backend/manager/models/session.py

```diff
@@ -1082,6 +1082,68 @@ async def get_sgroup_managed_sessions(
         return result.scalars().all()
 
 
+async def finalize_running(
```

Comment:
Most of the characteristics of this function are pretty simillar with those on `SessionRow.transit_session_status()`. I suppose we can just update implementations of the latter one instead of maintaining two duplicate functions.